### PR TITLE
[debugger] Added debugger support for Zephyr

### PR DESCRIPTION
**Pull request by jimmy-huang**
Created shim layer for building against the POSIX network stack on
Zephyr which enables the debugger to run on the frdm_k64f board with
TCP IPV4 on port 5001.

Signed-off-by: Jimmy Huang <jimmy.huang@intel.com>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -296,12 +326,9 @@ It will then be run on debugger mode waiting for client connection, you can then
 in another terminal, you can connect to it by running the python client in
 JerryScript:
 ```bash
-jerry-debugger/jerry-client-ws.py
+python jerryscript/jerry-debugger/jerry-client-ws.py --display 10 localhost
 ```
```
[the remark this answers is not in the document]
Comment (jimmy-huang) [reply README.md]: added

File: README.md [hunk 2]
```diff
@@ -282,12 +282,48 @@ breakpoints such as `b main` and `run` to start debugging as usual with gdb.
...
+
+Then you can use the jerryscript command line or html client to connect to the
+debugger to debug your JS application:
+
+python jerryscript/jerry-debugger/jerry-client-ws.py --display 10 192.168.1.1
+
+In the client, type 'help' to get a list of debugger commands, such as
+adding breakpoints, stepping through JS sources, etc.
+
+Alternatively, we've created a the client that integrates with Chome DevTools,
```
Comment (grgustaf) [inline README.md]: a client

File: README.md [hunk 3]
```diff
@@ -282,12 +282,48 @@ breakpoints such as `b main` and `run` to start debugging as usual with gdb.
...
+which you can use the Chrome's built-in debugger to connect and you can
```
Comment (grgustaf) [inline README.md]: s/you can use the/let's you use/

File: README.md [hunk 4]
```diff
@@ -282,12 +282,48 @@ breakpoints such as `b main` and `run` to start debugging as usual with gdb.
...
+use it to set breakpoints and step through source from all within the browser,
```
Comment (grgustaf) [inline README.md]: s/from//
browser. Please
Mr. Run-on Sentence :D

**Comment by grgustaf** [comment]
+1

**Pull request merged**